### PR TITLE
Handle non-existent accounts in preload

### PR DESCRIPTION
**Pull request by georgeee**
Problem: account preloading is not able to preserve information about some accounts not existing in ledger, which invites future repetitive lookups to DB during staged ledger application.

Solution: keep a set of non-existent accounts for masks populated on preload.

Impact: after preload underlying ledger is guaranteed not to be queried.

Explain how you tested your changes:
* [x] measured performance via #14582

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

File: src/lib/merkle_mask/masking_merkle_tree.ml [hunk 1]
```diff
@@ -274,34 +281,36 @@ module Make (Inputs : Inputs_intf.S) = struct
...
-              (rest, r)
-          | Some acc_found_locally, _ ->
-              (from_parent, (id, acc_found_locally))
-          | _ ->
-              failwith "unexpected number of results from DB" )
-      |> snd
+      if List.is_empty not_found then
+        List.map ~f:(fun (a, x) -> (a, Option.value_exn x)) self_found_or_none
+      else
+        let from_parent = lookup_parent ancestor not_found in
```
Comment (georgeee) [inline src/lib/merkle_mask/masking_merkle_tree.ml:287]: This code refactor is purely for clarity that call to parent happens only if some records were not found "locally" and underlying ledger needs to be queried.

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
A unit test is failing because of reparenting logic. This is a dead code (and a unit test for dead code) effectively. Still some code change is needed to either fix the dead code or remove it.

**Comment by mrmr1993** [comment]
> A unit test is failing because of reparenting logic. This is a unit test for dead code effectively. I.e. the code is used, but never in a way suggested by the test. Still some code change is needed to either fix the dead code or remove it.

The test is junk, feel free to just delete it.

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
> The test is junk, feel free to just delete it.

So the actual issue was not in reparenting logic, but in the fact that registering a mask happened before the last edit of the parent. This is something we don't do in production code, but were doing in the test and with the last commit of this PR it started failing the test because instead of calling the `last_filled` a `t.current_location` if the mask started to be used.

In near future I hope we rewrite the masking code so that it doesn't allow modification of a mask that has some children.

**Pull request merged**